sklearn/utils/estimator_checks.py

```diff
@@ -185,10 +185,12 @@ def _yield_regressor_checks(name, regressor):
 def _yield_transformer_checks(name, transformer):
     # All transformers should either deal with sparse data or raise an
     # exception with type TypeError and an intelligible error message
-    yield check_transformer_data_not_an_array
+    if not _safe_tags(transformer, "no_validation"):
```

Comment:
I think we should set the `no_validation` tag of `FunctionTransform` according to the `validate` param

Reply:
It is already set.

Reply:
one thing is to separate the test which is on the list and only run it on the "no_validation == False"

Reply:
Ok for the tag, but now I'm confused why the proposed changed introduce the need for `check_transformer_list`?

Reply:
checking the transformer without validation should work with an array but it could fail with a list.
The validation will make a check_array on the list and make it an array. So I thought that we could split these 2 cases to at least run the array test on the no validated transformer.

Reply:
But what is so different between `check_transformer_data_not_an_array` and your newly introduced `check_transformer_list`?

I would assume that `check_transformer_list` is redundant?

Reply:
I would say yes. But @amueller could have some clarifications regarding this case.

Reply:
Unless you can find a compelling reason to add the test, I don't see why we should have it

Reply:
I did move the check on the list with the `check_transformer_data_not_an_array` and remove the new test.
Is it ok with you @NicolasHug. I would like to keep this test on lists since it was already there.

Reply:
I understand the changes now, that makes sense. Thanks